framework/wazuh/engine/queries.py

```diff
@@ -0,0 +1,420 @@
+# Copyright (C) 2015, Wazuh Inc.
+# Created by Wazuh, Inc. <info@wazuh.com>.
+# This program is free software; you can redistribute it and/or modify it under the terms of GPLv2
+
+from typing import Any, Dict, List, Type
+
+from wazuh.core.exception import WazuhError
+from wazuh.core.utils import filter_array_by_query
+
+
+class BaseQuery:
+    """Base class for response transformations."""
+
+    def apply_transformation(self, data: List[Dict[str, Any]]):
+        """Apply the transformation to the given data.
+
+        Args:
+            data: The data to be transformed.
+
+        Raises:
+            NotImplementedError: This method must be implemented in the derived classes.
+        """
+        raise NotImplementedError
+
+    @staticmethod
+    def can_i_run(params: Dict[str, Any], data: Any) -> bool:
+        """Check if the transformation can be applied to the given parameters and data.
+
+        Args:
+            params: The parameters used for the transformation.
+            data: The data to be transformed.
+
+        Returns:
+            bool: True if the transformation can be applied, False otherwise.
+
+        Raises:
+            NotImplementedError: This method must be implemented in the derived classes.
+        """
```

Comment:
The docstring style should match the one used in the rest of the code:

```
Parameters
----------
params: Dict[str, Any]
    <description>

Returns
--------
bool
    <description>

Raises
------
WazuhError(<n>)
    <description>
```

Reply:
Changed all docstrings to the specified style in commit https://github.com/wazuh/wazuh/commit/7f5d9c70cdd5191ed427fecce30a54885ce8ae4e

Example:
https://github.com/wazuh/wazuh/blob/92f0a22c401a9cdc766321dfd3229823ae849dbb/framework/wazuh/core/engine/queries.py#L30-L50